### PR TITLE
Add deployment-templates directory

### PR DESCRIPTION
**Pull request by jkisk**
Adding a directory with sample goreleaser and release workflow files, along with a readme

Part of #https://github.com/pulumi/home/issues/1996

### CODE REVIEW

File: deployment-templates/.goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - CGO_ENABLED=0
+    goarch:
+      - amd64
+      - arm64
+    goos:
+      - darwin
+      - windows
+      - linux
+    ldflags:
+      # The linebelow MUST align with the module in current provider/go.mod
```
Comment (guineveresaenger) [inline deployment-templates/.goreleaser.yml:20]:
```suggestion
      # The line below MUST align with the module in current provider/go.mod
```

File: deployment-templates/release.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        uses: JS-DevTools/npm-publish@v1
+        with:
+          access: "public"
+          token: ${{ env.NPM_TOKEN }}
+          package: ${{github.workspace}}/sdk/nodejs/bin/package.json
+      - if: ${{ matrix.language == 'dotnet' && env.PUBLISH_NUGET == 'true' }}
+        name: publish nuget package
+        run: |
+          dotnet nuget push ${{github.workspace}}/sdk/dotnet/bin/Debug/*.nupkg -s ${{ env.NUGET_FEED_URL }} -k ${{ env.NUGET_PUBLISH_KEY }}
+          echo "done publishing packages"
```
Comment (guineveresaenger) [inline deployment-templates/release.yml:120]: I think we should add a `git tag sdk/` here, no?

File: deployment-templates/README-DEPLOYMENT.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+1. Customize .goreleaser.yml for your provider, paying special attention that the ldlflags are set to match your provider/go.mod exactly:
+
+`-X github.com/pulumi/pulumi-aws/provider/v5/pkg/version.Version={{.Tag}}`
+
+1. Delete this directory if desired
+
+
+# Deploy
+
+1. Push a tag to your repo in the format "v0.0.0" to initiate a release
```
Comment (stack72) [inline deployment-templates/README-DEPLOYMENT.md:26]: And also sdk/v0.0.0 for the Go SDK

File: deployment-templates/release.yml [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    - name: Install pulumictl
+      uses: jaxxstorm/action-install-gh-release@v1.5.0
+      with:
+        repo: pulumi/pulumictl
+    - name: Set PreRelease Version
+      run: echo "GORELEASER_CURRENT_TAG=v$(pulumictl get version --language generic)" >> $GITHUB_ENV
+    - name: Run GoReleaser
+      uses: goreleaser/goreleaser-action@v2
+      with:
+        args: -p 1 release --rm-dist --timeout 90m0s
```
Comment (stack72) [inline deployment-templates/release.yml:53]: We can change this to -p3 and remove the timeout as these are for AWS due to huge binary sizes

File: deployment-templates/release.yml [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        run: |
+          dotnet nuget push ${{github.workspace}}/sdk/dotnet/bin/Debug/*.nupkg -s ${{ env.NUGET_FEED_URL }} -k ${{ env.NUGET_PUBLISH_KEY }}
+          echo "done publishing packages"
+    strategy:
+      fail-fast: true
+      matrix:
+        dotnetversion:
+          - 3.1.301
+        goversion:
+          - 1.16.x
```
Comment (stack72) [inline deployment-templates/release.yml:127]: 1.17.x

**Review by stack72 — CHANGES_REQUESTED** [review]
I think there's a step that we are going to need to add to the Makefile for build_nodejs

when it's copying files we need to copy scripts to bin

**Review by stack72 — APPROVED** [review]

**Pull request merged**